src/jsonb_utils.c

```diff
@@ -57,6 +57,45 @@ ts_jsonb_add_str(JsonbParseState *state, const char *key, const char *value)
 	ts_jsonb_add_value(state, key, &json_value);
 }
 
+void
+ts_jsonb_set_value_by_type(JsonbValue *value, Oid typeid, Datum datum)
+{
+	switch (typeid)
+	{
+		Oid typeOut;
+		bool isvarlena;
+		char *str;
+
+		case INT8OID:
```

Comment:
Could you order the case statements? Now we have INT8,2,4

Reply:
Done. Also did another refactoring to make the code simpler to read.